instrumentation/hibernate/hibernate-4.0/javaagent/build.gradle.kts

```diff
@@ -32,6 +32,8 @@ dependencies {
 
   testImplementation("org.hibernate:hibernate-core:4.0.0.Final")
   testImplementation("org.hibernate:hibernate-entitymanager:4.0.0.Final")
+
+  testImplementation("org.javassist:javassist:3.28.0-GA")
```

Comment:
Is this used by hibernate?

Reply:
When I moved the `Value` class, the SpringJpa test started failing until I added this.

Here was the error:

>Error creating bean with name 'entityManagerFactory' defined in class spring.jpa.PersistenceConfig: Invocation of init method failed; nested exception is java.lang.RuntimeException: error trying to scan <jar-file>: file:/Users/jay/code/projects/opentelemetry-java-instrumentation/instrumentation/hibernate/hibernate-4.0/javaagent/build/classes/java/test/
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'entityManagerFactory' defined in class spring.jpa.PersistenceConfig: Invocation of init method failed; nested exception is java.lang.RuntimeException: error trying to scan <jar-file>: file:/Users/jay/code/projects/opentelemetry-java-instrumentation/instrumentation/hibernate/hibernate-4.0/javaagent/build/classes/java/test/
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1554)
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:539)
	at app//org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:475)
	at app//org.springframework.beans.factory.support.AbstractBeanFactory$1.getObject(AbstractBeanFactory.java:302)
	at app//org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:229)
	at app//org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:298)
	at app//org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:193)
	at app//org.springframework.context.support.AbstractApplicationContext.getBean(AbstractApplicationContext.java:975)
	at app//org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:752)
	at app//org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:482)
	at app//org.springframework.context.annotation.AnnotationConfigApplicationContext.<init>(AnnotationConfigApplicationContext.java:84)
	at app//SpringJpaTest.$spock_initializeSharedFields(SpringJpaTest.groovy:21)
Caused by: java.lang.RuntimeException: error trying to scan <jar-file>: file:/Users/jay/code/projects/opentelemetry-java-instrumentation/instrumentation/hibernate/hibernate-4.0/javaagent/build/classes/java/test/
	at org.hibernate.ejb.Ejb3Configuration.scanForClasses(Ejb3Configuration.java:855)
	at org.hibernate.ejb.Ejb3Configuration.configure(Ejb3Configuration.java:597)
	at org.hibernate.ejb.HibernatePersistence.createContainerEntityManagerFactory(HibernatePersistence.java:72)
	at org.springframework.orm.jpa.LocalContainerEntityManagerFactoryBean.createNativeEntityManagerFactory(LocalContainerEntityManagerFactoryBean.java:338)
	at org.springframework.orm.jpa.AbstractEntityManagerFactoryBean.afterPropertiesSet(AbstractEntityManagerFactoryBean.java:318)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1613)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1550)
	... 11 more
Caused by: java.lang.RuntimeException: Error while reading file:/Users/jay/code/projects/opentelemetry-java-instrumentation/instrumentation/hibernate/hibernate-4.0/javaagent/build/classes/java/test/
	at org.hibernate.ejb.packaging.NativeScanner.getFilesInJar(NativeScanner.java:193)
	at org.hibernate.ejb.Ejb3Configuration.addScannedEntries(Ejb3Configuration.java:504)
	at org.hibernate.ejb.Ejb3Configuration.scanForClasses(Ejb3Configuration.java:852)
	... 17 more
Caused by: java.io.IOException: invalid constant type: 18
	at javassist.bytecode.ConstPool.readOne(ConstPool.java:1090)
	at javassist.bytecode.ConstPool.read(ConstPool.java:1033)
	at javassist.bytecode.ConstPool.<init>(ConstPool.java:149)
	at javassist.bytecode.ClassFile.read(ClassFile.java:737)
	at javassist.bytecode.ClassFile.<init>(ClassFile.java:108)
	at org.hibernate.ejb.packaging.AbstractJarVisitor.checkAnnotationMatching(AbstractJarVisitor.java:245)
	at org.hibernate.ejb.packaging.AbstractJarVisitor.executeJavaElementFilter(AbstractJarVisitor.java:211)
	at org.hibernate.ejb.packaging.AbstractJarVisitor.addElement(AbstractJarVisitor.java:172)
	at org.hibernate.ejb.packaging.ExplodedJarVisitor.getClassNamesInTree(ExplodedJarVisitor.java:125)
	at org.hibernate.ejb.packaging.ExplodedJarVisitor.getClassNamesInTree(ExplodedJarVisitor.java:133)
	at org.hibernate.ejb.packaging.ExplodedJarVisitor.getClassNamesInTree(ExplodedJarVisitor.java:133)
	at org.hibernate.ejb.packaging.ExplodedJarVisitor.getClassNamesInTree(ExplodedJarVisitor.java:133)
	at org.hibernate.ejb.packaging.ExplodedJarVisitor.getClassNamesInTree(ExplodedJarVisitor.java:133)
	at org.hibernate.ejb.packaging.ExplodedJarVisitor.getClassNamesInTree(ExplodedJarVisitor.java:133)
	at org.hibernate.ejb.packaging.ExplodedJarVisitor.getClassNamesInTree(ExplodedJarVisitor.java:133)
	at org.hibernate.ejb.packaging.ExplodedJarVisitor.doProcessElements(ExplodedJarVisitor.java:91)
	at org.hibernate.ejb.packaging.AbstractJarVisitor.getMatchingEntries(AbstractJarVisitor.java:148)
	at org.hibernate.ejb.packaging.NativeScanner.getFilesInJar(NativeScanner.java:190)
	... 19 more